packages/babel-parser/src/types.js

```diff
@@ -956,7 +956,12 @@ export type TsTypeParameterDeclaration = TypeParameterDeclarationBase & {
 };
 
 export type TypeParameterBase = NodeBase & {
+  // TODO(Babel-8): remove name property
+  /**
+   * @deprecated use `typeName` instead
+   */
   name: string,
+  typeName: Identifier,
```

Comment:
We can also handle this as a union type assigning it under the `BABEL_8_BREAKING` flag.
```suggestion
  name: string | Identifier,
```